src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,153 @@
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
+interface Props {
+  selectedPostId: number;
+}
+
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [post, setPost] = useState<Post | null>(null);
+  const [isPostLoading, setIsPostLoading] = useState(false);
+  const [comments, setComments] = useState<PostComment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [isCommentsShow, toggleIsCommentsShow] = useState(true);
+
+  const loadPostById = async (postId: number) => {
+    try {
+      setIsPostLoading(true);
+      const postsFromServer = await getPostDetails(postId);
+
+      setPost(postsFromServer);
+    } catch (error) {
+      throw new Error(`${error}`);
+    } finally {
+      setIsPostLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPostById(selectedPostId);
+  }, [selectedPostId]);
+
+  const loadCommentsByPostId = async (postId: number) => {
+    try {
+      setIsCommentsLoading(true);
+      const commentsFromServer = await getPostComments(postId);
+
+      setComments(commentsFromServer);
+    } catch (error) {
+      throw new Error(`${error}`);
+    } finally {
+      setIsCommentsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadCommentsByPostId(selectedPostId);
+  }, [selectedPostId]);
+
+  const handleRemoveCommentById = async (commentId: number, postId: number) => {
+    try {
+      setIsCommentsLoading(true);
+      await removeComment(commentId);
+      const commentsFromServer = await getPostComments(postId);
+
+      setComments(commentsFromServer);
+    } catch (error) {
+      throw new Error(`${error}`);
+    } finally {
+      setIsCommentsLoading(false);
+    }
+  };
+
+  const handleAddComment = async (newComment: NewComment) => {
+    try {
+      setIsCommentsLoading(true);
+      await addComment(newComment);
+      const commentsFromServer = await getPostComments(newComment.postId);
+
+      setComments(commentsFromServer);
+    } catch (error) {
+      throw new Error(`${error}`);
+    } finally {
+      setIsCommentsLoading(false);
+    }
```

Comment:
Fix this in all places
```suggestion
    } catch (error) {
      setIsCommentsLoading(false);
      throw new Error(`${error}`);
    }
```

Reply:
the same as before